test/functional/rpc_getversion.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the getversion RPC."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+
+class GetVersionTest (BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def run_test(self):
+        version = self.nodes[0].getversion()
+        # The functional tests don't have access to the version number and it
+        # can't be hardcoded. Only doing sanity and consistency checks in this
+        # test.
```

Comment:
You could do this by inspecting the logs?

<details>
<summary>git diff on 54cb1abffa</summary>

```diff
diff --git a/test/functional/rpc_getversion.py b/test/functional/rpc_getversion.py
index 712c1f76fc..366ca378e9 100755
--- a/test/functional/rpc_getversion.py
+++ b/test/functional/rpc_getversion.py
@@ -4,6 +4,8 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 """Test the getversion RPC."""
 
+import re
+
 from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import assert_equal
 
@@ -11,12 +13,25 @@ from test_framework.util import assert_equal
 class GetVersionTest (BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
+        
+    def parse_version_from_log(self):
+        version_pattern = re.compile(r"Bitcoin Core version (\S+)")
+        
+        with open(self.nodes[0].debug_log_path, 'r') as file:
+            for line in file:
+                match = version_pattern.search(line)
+                if match:
+                    return match.group(1)
+        
+        assert False, "No version found in log file."
 
     def run_test(self):
+        log_version = self.parse_version_from_log()
         version = self.nodes[0].getversion()
-        # The functional tests don't have access to the version number and it
-        # can't be hardcoded. Only doing sanity and consistency checks in this
-        # test.
+        import pdb; pdb.set_trace()
+        
+        # compare log output and RPC output
+        assert_equal(log_version, version["long"])
 
         # numerical: e.g. 279900
         assert_equal(type(version["numeric"]), int)

```
</details>

Reply:
I agree that this would work. I'm a bit reluctant to introduce more log parsing for tests. I'm leaving unresolved: If someone feels strongly about doing this, let me know. 

An alternative I have though about is adding the required version information to `test/config.ini`. Again, also open to do this if someone thinks it's important to do it.

Reply:
Was talking with @pinheadmz about this. Maybe as an alternative, the test could fetch the `getversion` RPC response (e.g. with "27.0..."), then use `assert_debug_log()` to check that the RPC response matches what's in the log.